compiler/lib/linker.ml

```diff
@@ -396,20 +416,45 @@ let load_fragment ~filename f =
                 let named_values = find_named_value code in
                 Primitive.register name kind ka arity;
                 StringSet.iter Primitive.register_named_value named_values;
-                (if Hashtbl.mem provided name
-                then
-                  let _, ploc, weakdef = Hashtbl.find provided name in
-                  if not weakdef
+                let is_updating =
+                  if Hashtbl.mem provided name
                   then
-                    warn
-                      "warning: overriding primitive %S\n  old: %s\n  new: %s@."
-                      name
-                      (loc ploc)
-                      (loc pi));
-                Hashtbl.add provided name (id, pi, weakdef);
-                Hashtbl.add provided_rev id (name, pi);
-                check_primitive ~name pi ~code ~requires;
-                Hashtbl.add code_pieces id (code, requires)))
+                    let _, ploc, weakdef, prev_target_env = Hashtbl.find provided name in
+                    if not weakdef
+                    then (
+                      match prev_target_env, target_env with
+                      | `Isomorphic, `Browser -> Config.Flag.is_targetting_browser_env ()
+                      | `Isomorphic, `Nodejs -> Config.Flag.is_targetting_nodejs_env ()
+                      | `Nodejs, `Browser
+                      | `Browser, `Nodejs
+                      | `Nodejs, `Isomorphic
+                      | `Browser, `Isomorphic ->
```

Comment:
The logic here depends on the order of the stubs / files. We could also take another approach and take the most precise version. What do you think ?

Reply:
ya, i considered it earlier on. i kind of gave it a 🤷 as perhaps bikeshedding in my first pass. but nonetheless, it's easy to do, so i sent the required patch in 97f1e42a5c